spec-examples/src/chacha20_poly1305/chacha20.rs

```diff
@@ -84,16 +84,20 @@ pub fn block(key: Key, ctr: U32, iv: IV) -> StateBytes {
     state_to_bytes(state)
 }
 
-pub fn chacha(key: Key, iv: IV, m: &ByteSeq) -> Result<ByteSeq, String> {
-    let mut ctr = U32(1);
+pub fn chacha(key: Key, iv: IV, m: &ByteSeq) -> ByteSeq {
+    let mut ctr = U32(1u32);
     let mut blocks_out = ByteSeq::new(m.len());
-    for i in 0..m.num_chunks(64) {
-        let (block_len, msg_block) = m.get_chunk(64, i);
+    for i in 0usize..m.num_chunks(64usize) {
+        let (block_len, msg_block) = m.get_chunk(64usize, i);
         let key_block = block(key, ctr, iv);
         let msg_block_padded = StateBytes::new();
         let msg_block_padded = msg_block_padded.update_start(&msg_block);
-        blocks_out = blocks_out.set_chunk(64, i, &(msg_block_padded ^ key_block).slice_range(0..block_len));
-        ctr += U32(1);
+        blocks_out = blocks_out.set_chunk(
+            64usize,
+            i,
+            &(msg_block_padded ^ key_block).slice_range(0usize..block_len),
+        );
+        ctr = ctr + U32(1u32);
     }
-    Ok(blocks_out)
```

Comment:
hm, why was this a `Result` before ... ?

Reply:
I don't know. Chacha20 always succeeds so I don't know why there was a `Result` there before.